### DISCUSSION TITLE
Open to add closed for update

### DISCUSSION THREAD

**Discussion by ArvindGundimajalu** [git-for-windows/git#4804] [Ideas]
I am looking for a feature that allows me to lockdown updates to files. Essentially making them immutable.
This will be really handy for files that define interface, allowing to control the possibility of a breaking change getting introduced.
The feature needs to operate at 2 levels 
1. File level - allowing to lock specific files.
2. Folder level - allowing to lock all files under a specific path.

The feature should now block a user from adding new files. But once added it should not be possible to update or delete the file without special privileges.

**Comment by tboegi** [comment]
Git itself is not designed to do this kind of locking.
In opposite to other version control systems like rcs
(Or should I say revision control systems ?)
Git checkouts files always as writable/changeable.
git status or git diff is your friend here.

However, when you really want to set a file to be "read only",
you can do this with windows tools.


https://en.wikipedia.org/wiki/Version_control
